include/wx/generic/maskededit.h

```diff
@@ -0,0 +1,681 @@
+    {
+        return m_params;
+    }
+
+    //Event handling
+    void OnKeyDown(wxKeyEvent& event);
+    void OnChar(wxKeyEvent& event);
+    void OnCut(wxCommandEvent& event);
+    void OnPaste(wxCommandEvent& event);
+#if wxUSE_COMBOBOX
+    void OnComboSelect(wxCommandEvent& event);
+    void OnComboDrop(wxCommandEvent& event);
+#endif // wxUSE_COMBOBOX
+
+protected:
+    //Like SetFieldValue(), but operates on inText, not in control
+    wxString SetFieldOnText(size_t index,
+                            const wxString& plainValue,
+                            const wxString& inText);
+
+    //Like GetFieldValue(), but operates on inText, not in control
+    wxString GetFieldOnText(size_t index,
+                            const wxString& inText,
+                            wxChar chPad) const;
+
+    //If the cell has not been filled
+    bool CellIsEmpty(long curpos) const
+    {
+        return !m_filled[(size_t)curpos];
+    }
+
+    //Get the internal type for a char in the expanded mask
+    int GetMaskedType(size_t pos) const
+    {
+        return m_maskIDs[pos];
+    }
+
+    //Default colours
+    void GetDefaultColours(wxColour* defFgCol, wxColour* defBgCol);
+
+    //Handle the selection, returning the new possible control's value
+    wxString HandleSelection(long *pos, bool *changed);
+
+    //Our control derives from wxTextEntry, but not this class. Get a pointer to it
+    wxTextEntry* GetTextEntry() const;
+
+    //Update colours, depending on empty/valid/invalid cases
+    void UpdateMaskedColours();
+
+    //Join characters and align the field
+    void NormalizeField(size_t index, wxString* inText);
+
+    //Test if the field has a free cell for a char.
+    bool HasFieldRoom(size_t index) const;
+
+    //Find the field in which the cursor is.
+    int FindField(long curpos) const;
+    //Find the field to insert a char in the given position.
+    int GuessField(long pos) const;
+
+    //A reasonable position for caret in the previous field to where the caret is.
+    long PosForFieldPrev(long curpos) const;
+    //A reasonable position for caret in the next field to where the caret is.
+    long PosForFieldNext(long curpos) const;
+
+    //Move caret from current position to the foreseeable one
+    long MoveCaret(long curPos, bool toRight);
+
+    //Insert a char at an intended position. Returns false if insertion is not
+    // possible (invalid char or no room available)
+    bool InsertAtPos(wxChar iChar, long* pos, size_t index, wxString* inText);
+
+    //Process a char, coming from OnChar() or from DoPaste()
+    bool ProcessChar(wxChar aChar, long* pos, wxString* inText, bool* changed);
+
+    //Cutting
+    void DoCut();
+    //Pasting
+    void DoPaste();
+
+    //Set the proper string for SetValue/ChangeValue
+    wxString SetMaskedValue(const wxString& value);
+
+    //Show the colours
+    void UseColours(int whatColours);
+
+    enum
+    {
+        mEmColour = 1,
+        mOKColour,
+        mWRColour
```

Comment:
Naming convention again. Also, we don't use leading `m` for non-member variables.